### DISCUSSION TITLE
ACE_QtReactor lib not being created with OpenDDS 3.16 configure script

### DISCUSSION THREAD

**Discussion by hmcrowder** [OpenDDS/OpenDDS#2489] [Q&A]
Using OpenDDS-3.16 on a linux VM (RHEL 7.8) with Qt5 installed.  The ACE_QtReactor is not getting built.
Used the following configure script: ./configure --prefix=/mypath/ --qt=/usr/local/bin/Qt-5.12.4/ --qt-include=/mypath/qt/include
NOTE: I've tried many variants of such and I am unable to identify what needs to be done to have it installed with the ACE_QtReactor lib files included.

**Comment by iguessthislldo** [comment]
OpenDDS doesn't use `ACE_QtReactor` anywhere and it's an optional part of ACE, so when the configure script is allowed to build ACE it doesn't include it. If you want to build it you'll have to build ACE separately as you wish or add `$(ACE_ROOT)/ace/QtReactor/ace_qt5reactor.mpc` to [`ACE_TAO_for_OpenDDS.mwc`](https://github.com/objectcomputing/OpenDDS/blob/master/ACE_TAO_for_OpenDDS.mwc).

**Reply by iguessthislldo** [reply]
Alternatively if you already built OpenDDS, you can go to `$ACE_ROOT/ace/QtReactor` and build it separately since it's a separate library.

**Reply by mitza-oci** [reply]
Or build ACE however you want to before starting to build OpenDDS, and point OpenDDS's configure to that built ACE.

**Reply by hmcrowder** [reply]
I must be missing a point here, because I put the ace_qt5reactor.mpc into the ACE_TAO_for_OpenDDS.mwc.  This didn't build the libACE_QtReactor library.  I also tried to just run the make on the $ACE_ROOT/ace/QtReactor, and it's not creating the library file.  It states it will not be built due to the following disabled make macros: qt5.  What needs to be done to enable this macro?  NOTE:  I verified that Qt version 5.12.4 is installed.

**Reply by hmcrowder** [reply]
I created an env var qt5=1 and that allowed me to build the ACE_QtReactor.  However, is this really the proper way of handling this - or is there some other way of handling this?

**Reply by mitza-oci** [reply]
If you are configuring ACE separately, make sure to update the platform_macros.GNU file.
However it is not clear to me if you are configuring ACE separately.

**Reply by hmcrowder** [reply]
I'd prefer to not configure ACE separately.  I am trying to just use the instructions provided to configure script for OpenDDS, so that it supports Qt5 and supporting QtReactor.  Since the configure script states that it will pull down the TAO that is needed - I did not want to explicitly configure anything if possible.   So - I followed the advice provided initially and added the QtReactor to the ACE_TAO_for_OpenDDS.mwc, but although it doesn't "skip" the qt5, it also didn't create the library because it stated that the qt5 macro was not enabled.  In regards to the platform_macros.GNU file.   I was under the impression that setting the parameters in the configure would update what is needed (and it does add the prefix to the platform_macros.GNU file), but it does not add anything about qt.  What should be added to the platform_macros.GNU?
